aslam_cv_cameras/src/camera.cc

```diff
@@ -59,37 +63,175 @@ void Camera::printParameters(std::ostream& out, const std::string& text) const {
   out << "  image (cols,rows): " << imageWidth() << ", " << imageHeight() << std::endl;
 }
 
-bool Camera::operator==(const Camera& other) const {
-  // \TODO(slynen) should we include the id and name here?
+bool Camera::isEqualCameraImpl(const Camera& other) const {
   return (this->intrinsics_ == other.intrinsics_) &&
```

Comment:
are the other checks evaluated if the first one fails?

Reply:
No, `&&` will stop evaluating the moment it encounters an expression that evaluates to false.